src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,106 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
-    public int size() {
-        return 0;
+    private ArrayList<T> elementData  = new ArrayList<>();
+    private T to;
+    private T from;
+
+    public Range(T from, T to, Function<T, T> function) {
+        this.from = from;
+        this.to = to;
+        if(!(from.equals(to))) {
+
+            do {
+                this.elementData.add(from);
+                from = function.apply(from);
+            }
+            while (!(from.equals(to)));
+            this.elementData.add(to);
+        }
+    }
+
+
+    public static Range<Integer> of(Integer from, Integer to) {
+        return new Range<>(from, to, intValue -> intValue + 1);
+    }
+
+    public static Range<Float> of(Float from, Float to) {
+        return new Range<>(from, to, floatValue -> floatValue + 0.1f);
     }
 
-    public boolean isEmpty() {
-        return false;
+    public static Range<Character> of(char from, char to, Function<Character, Character> characterCharacterFunction) {
+        return new Range<>(from, to, characterCharacterFunction);
     }
 
+    public int size() {
+        return this.elementData.size();
+    }
+
+    public boolean isEmpty() { return this.elementData.isEmpty();}
+
     public boolean contains(Object o) {
-        return false;
+        return this.elementData.indexOf(o) >= 0;
     }
 
     public Iterator<T> iterator() {
-        return null;
+        Iterator iterator = this.elementData.iterator();
+        return iterator;
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        return this.elementData.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+        return this.elementData.toArray(a);
     }
 
+
     public boolean add(T t) {
-        return false;
+
+        if(this.elementData.indexOf(t) >= 0){
+
+            return false;
+        }
+
+        this.elementData.add(t);
+        return true;
     }
 
     public boolean remove(Object o) {
-        return false;
+
+        return  this.elementData.remove(o);
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        return this.elementData.containsAll(c);
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+        for (T element:c) {
+            if(!(this.elementData.add(element))){
+                return false;
```

Comment:
You have to add ALL elements from **c**, if at least one was added then return true.